### PR TITLE
Alerting: fix alert groups grouping

### LINKED ISSUE #39674: Alerting UI: Grouping appears to be off
It seems that with the default of "no grouping" e.g. `group_by: ""` the "alert groups" tab gets confused. 

It shows something like:

Screenshot 2021-09-23 at 10 18 53

But I don't understand a) Why is there a separate group at the top and b) Why is there a `group by` empty header - should we say something like `no grouping` specified and make it all one big group?

Perhaps this is a bug?

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

If there are two notification policies that have the same set of label matchers, then the Grafana alertmanager will return two separate groupings for the alerts even though they both have empty labels that they're grouped by. This resulted in multiple "No grouping" groups in the UI. This combines them all into one.

**Which issue(s) this PR fixes**:

Fixes #39674 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gillesdemey** [comment]
I'm wondering if we should also omit the duplicated alert instances; it seems a bit weird to render them multiple times – any thought on that?

image

**Comment by nathanrodman** [comment]
> I'm wondering if we should also omit the duplicated alert instances; it seems a bit weird to render them multiple times – wdyt?
> 
> image

That sounds reasonable, I think the duplicate alerts will have different receivers but we can merge them together

**Comment by gillesdemey** [comment]
CI failure seems unrelated; restarting

File: public/app/features/alerting/unified/hooks/useGroupedAlerts.ts [hunk 1]
```diff
@@ -1,11 +1,30 @@
 import { useMemo } from 'react';
 import { AlertmanagerGroup } from 'app/plugins/datasource/alertmanager/types';
 import { Labels } from '@grafana/data';
+import { uniqBy } from 'lodash';
 
-export const useGroupedAlerts = (groups: AlertmanagerGroup[], groupBy: string[]) => {
+export const useGroupedAlerts = (groups: AlertmanagerGroup[], groupBy: string[]): AlertmanagerGroup[] => {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/hooks/useGroupedAlerts.ts:6]: Can we write a unit test for the `useGroupedAlerts` function? It's getting a bit complex and a unit test could help us refactor it later

Comment (nathanrodman) [reply public/app/features/alerting/unified/hooks/useGroupedAlerts.ts:6]: Agreed

**Review by gillesdemey — APPROVED** [review]
LGTM! 👏

**Comment by Elfo404** [comment]
bumping this to 8.4.4 as it's not currently backported

**Pull request merged**